internal/ld/ld.go

```diff
@@ -74,6 +74,32 @@ func IsTransient(err error) bool {
 	return !errors.As(err, &e)
 }
 
+// LaunchDarkly API uses the X-Ratelimit-Reset header to communicate when to retry after a 429
+// Fallback to default backoff if header can't be parsed
+// https://apidocs.launchdarkly.com/#section/Overview/Rate-limiting
+// Method is curried in order to avoid stubbing the time package in unit tests
+func RateLimitBackoff(now func() time.Time) func(time.Duration, time.Duration, int, *http.Response) time.Duration {
+	return func(min, max time.Duration, attemptNum int, resp *http.Response) time.Duration {
+		if resp != nil {
+			if resp.StatusCode == http.StatusTooManyRequests {
+				if s, ok := resp.Header["X-Ratelimit-Reset"]; ok {
+					if sleepUntil, err := strconv.ParseInt(s[0], 10, 64); err == nil {
+						sleep := sleepUntil - now().UnixMilli()
+
+						if sleep > 0 {
+							return time.Millisecond * time.Duration(sleep)
+						} else {
+							return time.Duration(0)
+						}
+					}
+				}
+			}
+		}
+
+		return h.DefaultBackoff(min, max, attemptNum, resp)
```

Comment:
Can you update this to previous behavior, which has `h.LinearJitterBackoff` as the default?

Reply:
Done in 7d9f31f07a75805587000521a7b04963d42068e7. `h.DefaultBackoff` is still used for the unit test since `h.LinearJitterBackoff` depends on `math/rand`.